Lib/multiprocessing/pool.py

```diff
@@ -429,6 +450,17 @@ def _map_async(self, func, iterable, mapper, chunksize=None, callback=None,
         )
         return result
 
+    def _wait_for_updates(self, timeout=None):
+        task_queue_sentinels = [self._outqueue._reader]
+        self_notifier_sentinels = [self._change_notifier._reader]
+        worker_sentinels = [worker.sentinel for worker in self._pool]
+        sentinels = [*task_queue_sentinels,
+                     *worker_sentinels,
+                     *self_notifier_sentinels]
+        wait(sentinels, timeout=timeout)
+        while not self._change_notifier.empty():
+            self._change_notifier.get()
```

Comment:
I don't think that this code is safe, it looks like a race condition: https://en.wikipedia.org/wiki/Time_of_check_to_time_of_use

I suggest to call get(block=False) in a loop until you get an Empty exception.

Note: the race condition is not really critical, since it's fine if we miss a few events.

Reply:
There is no race condition as long as this is the only thread that pops from the queue.

Reply:
Also, sadly `queue.SimpleQueue` has no `block=False` option (we could add one, but I think is not needed).